config/config.go

```diff
@@ -84,8 +85,9 @@ func NewConfig() *Config {
 
 	// Set any defaults which do not align with Go zero values.
 	var initParsers = []configParser{defaultPullModes, defaultDirectoryCacheConfig}
-	for _, p := range append(initParsers, parsers...) {
-		p(cfg)
+	err := parseConfig(cfg, append(initParsers, parsers...))
+	if err != nil {
+		return nil
 	}
```

Comment:
nit: we're mixing just assigning to err and the "if scoped" error handling patterning in this code for no reason. Since error handling already proved problematic here, can we just make all of the error handling use the if pattern?

```suggestion
	if err := parseConfig(cfg, append(initParsers, parsers...)); err != nil {
		return nil
	}
```